clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py

```diff
@@ -97,11 +100,20 @@ def analyze_samples(self, positive_controls, negative_controls, samples):
             self._analyze_controls(positive_controls, negative_controls)
         except MultipleAnalysisErrors:
             for sample in samples:
-                yield {"id": sample["id"], "diagnosis_result": COVID_RESPONSE_FAILED}
+                yield {"id": sample["id"], DIAGNOSIS_RESULT: COVID_RESPONSE_FAILED}
             # No need to continue looking at the samples, as we fail them all here.
             return
 
         # Check samples
         for sample in samples:
             result = self._analyze_sample(sample)
-            yield {"id": sample["id"], "diagnosis_result": result}
+            yield {"id": sample["id"], DIAGNOSIS_RESULT: result}
+
+
+class CTResult(namedtuple('CTResult', ['id', 'fam_ct', 'hex_ct'])):
```

Comment:
The contract is a bit more complicated than this, since FAM and HEX are not looked properties, but can vary based on machine and kit types from what I understand.

Reply:
You can see a bit more of what I mean here: https://github.com/ctmrbio/claritylims/pull/43

Reply:
My intention was to have a contract in the same file as the implementation, to make it more clear to callers when it changes.